spring-ws-core/src/test/java/org/springframework/ws/server/endpoint/observation/ObservationInterceptorIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * Verifies observation for a WebService Endpoint.
+ * @author Johan Kindgren
+ */
+public class ObservationInterceptorIntegrationTest {
+
+    private static AnnotationConfigWebApplicationContext applicationContext;
+    private WebServiceTemplate webServiceTemplate;
+    private TestObservationRegistry registry;
+    private static Server server;
+
+    private final String requestPayload = "<root xmlns='http://springframework.org/spring-ws'><child/></root>";
+
+    private TransformerFactory transformerFactory = TransformerFactory.newInstance();
+    private Transformer transformer;
+
+    private static String baseUrl;
+
+    @BeforeAll
+    public static void startServer() throws Exception {
+
+        int port = FreePortScanner.getFreePort();
+
+        baseUrl = "http://localhost:" + port + "/ws";
+
+        server = new Server(port);
+        ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
+        context.setContextPath("/");
+
+        applicationContext = new AnnotationConfigWebApplicationContext();
+        applicationContext.scan(WebServiceConfig.class.getPackage().getName());
+
+        MessageDispatcherServlet dispatcherServlet = new MessageDispatcherServlet(applicationContext);
+        dispatcherServlet.setTransformWsdlLocations(true);
+
+        ServletHolder servletHolder = new ServletHolder(dispatcherServlet);
+        context.addServlet(servletHolder, "/ws/*");
+
+        server.setHandler(context);
+        server.start();
+    }
+
+    @AfterAll
+    static void tearDown() throws Exception {
+        applicationContext.close();
+        server.stop();
+    }
+
+    @BeforeEach
+    void setUp() throws TransformerConfigurationException {
+
+        webServiceTemplate = applicationContext.getBean(WebServiceTemplate.class);
+        registry = applicationContext.getBean(TestObservationRegistry.class);
+
+        transformer = transformerFactory.newTransformer();
+    }
+
+    @Test
+    void testObservationInterceptorBehavior() {
+
+        MyEndpoint.MyRequest request = new MyEndpoint.MyRequest();
+        request.setName("John");
+        MyEndpoint.MyResponse response = (MyEndpoint.MyResponse) webServiceTemplate.marshalSendAndReceive(baseUrl, request);
+
+        // Assertions based on expected behavior of ObservationInterceptor
+        assertNotNull(response);
+
+        assertThat(registry).hasAnObservation(observationContextAssert ->
+                observationContextAssert
+                        .hasLowCardinalityKeyValue("outcome", "success")
+                        .hasLowCardinalityKeyValue("exception", "none")
+                        .hasLowCardinalityKeyValue("namespace", "http://springframework.org/spring-ws")
+                        .hasLowCardinalityKeyValue("localpart", "request")
+                        .hasLowCardinalityKeyValue("soapaction", "none")
+                        .hasContextualNameEqualTo("POST /ws")
```

Comment:
What is the `path` key-value in this test case?

Reply:
If the current request path fully matches the registered servlet path, there will be no varying high cardinality key. Would it be better to add a 'none'-value?

Reply:
I think I misunderstood your comment, I have now added the path-key to the observation. The high cardinality key 'pathInfo' is only provided when the request has a wildcard match.

Reply:
My comment could have been more clear. I was expecting if there is a `path` key-value that it would be in the assertions here, but it isn't.

Reply:
I added the path key to the test, but it doesn't seem to show in this view?